data_process.py

```diff
@@ -406,7 +411,96 @@ def augment(data, augment_func=add_noise_to_data_point, augment_proportion=0.5,
     
     return
 
+
+def get_specific_date_data(date_data, data, start_date='2000-01-01', end_date='2020-01-01'):
+    '''
+    Parameters
+    ----------
+    date_data : numpy array dtype '<U10'
+        shape (N,1)
+    data : numpy array dtype 'float32'
+        shape (N,M)
+    start_date : string date 'YYYY-mm-dd'
+    end_date : string date 'YYYY-mm-dd'
+
+    Returns
+    -------
+    data_in_range : list of tuples (x,t)
+        x - shape (x_length, M)
+        t - shape (t_length, M)
+    '''
+    N = data.shape[0]
+    start_date = datetime.datetime.strptime(start_date, '%Y-%m-%d')
+    end_date = datetime.datetime.strptime(end_date, '%Y-%m-%d')
     
+    # start, end indexes corresponding to dates
+    start_index = -1
+    end_index = -1
+    for i in range(N):
+        date_i = datetime.datetime.strptime(date_data[i,0], '%Y-%m-%d')
+        
+        # train dates
+        if date_i >= start_date and start_index == -1:
+            start_index = int(i)
+        elif date_i >= end_date and end_index == -1:
+            end_index = int(i)
+    
+    data_in_range = data[start_index:end_index]
+
+    return data_in_range
+
+
+def normalize_single_stock(data_date):
```

Comment:
Can we rename to data for clarity? This function doesn't seem to handle dates like the others.
```suggestion
def normalize_single_stock(data):
```